File: Python/Mahasiswa.py
from Akademik import Akademik

class Mahasiswa(Akademik):
    __nim = ""
    __prodi = ""
    __fakultas = ""

    def __init__(self, kampus="", email="", nim ="", prodi="", fakultas=""):
        super().__init__(kampus, email)
        self.__nim = nim
        self.__prodi = prodi
        self.__fakultas = fakultas
    
    # Getter
    # NIM
    def getNim(self):
        return self.__nim
    
    # Prodi
    def getProdi(self):
        return self.__prodi
    
    # Fakultas
    def getFakultas(self):
        return self.__fakultas
    
    # Setter
    # NIM
    def setNim(self, nim):
        self.__nim = nim
    
    # Prodi
    def setProdi(self, prodi):
        self.__prodi = prodi
    
    # Fakultas
    def setFakultas(self, fakultas):
        self.__fakultas = fakultas

File: C++/Mahasiswa.cpp
class Mahasiswa : public Akademik{
    private :
        string nim;
        string prodi;
        string fakultas;

    public :

        Mahasiswa(){
        }

        /* Setter and Getter NIM*/
        void setNIM(string nim){
            this->nim = nim;
        }

        string getNIM(){
            return this->nim;
        }

        /* Setter and Getter Prodi*/
        void setProdi(string prodi){
            this->prodi = prodi;
        }

        string getProdi(){
            return this->prodi;
        }

        /* Setter and Getter Fakultas */
        void setFakultas(string fakultas){
            this->fakultas = fakultas;
        }

        string getFakultas(){
            return this->fakultas;
        }

        ~Mahasiswa(){

        }
};